nimterop/getters.nim

```diff
@@ -306,4 +313,27 @@ proc getNimExpression*(expr: string): string =
 
 proc getSplitComma*(joined: seq[string]): seq[string] =
   for i in joined:
-    result = result.concat(i.split(","))
\ No newline at end of file
+    result = result.concat(i.split(","))
+
+proc dll*(path: string): string =
+  let
+    (dir, name, ext) = path.splitFile()
+
+  when defined(Windows):
+    result = dir/name.addFileExt("dll")
+  when defined(Linux):
+    result = dir/"lib" & name.addFileExt("so")
+  when defined(OSX):
+    result = dir/"lib" & name.addFileExt("dylib")
+
+proc loadPlugin*(fullpath: string) =
+  doAssert fileExists(fullpath), "Plugin file does not exist: " & fullpath
+  if not fileExists(fullpath.dll):
```

Comment:
Plugin code is completely in the file and 100% Nim, why will -D impact it?

I agree we want to disable caching though. Have added some code to that effect.

Reply:
`-DFOO` was a bad example; but adding stuff to `~/.config/nim/config.nim` can influence behavior of that plugin; so ya that `gStateCT.nocache` u added shd take care of this

[RESOLVED]

Reply:
so i guess same thing here? `compileOption("forceBuild")` ? but maybe factorize so `compileOption("forceBuild")` is called just once in case we wanna change that logic eg: `const forceBuild = compileOption("forceBuild")` and then use `forceBuild` in the right places, i guess force recompilation of plugins as well as the other things ( c sources?)